samples/Humidity_BME280/app/application.cpp

```diff
@@ -0,0 +1,83 @@
+/***************************************************************************
+ * Sming port of example from Adafruit_BME280_Library
+
+  This is a library for the BME280 humidity, temperature & pressure sensor
+
+  Designed specifically to work with the Adafruit BME280 Breakout
+  ----> http://www.adafruit.com/products/2650
+
+  These sensors use I2C or SPI to communicate, 2 or 4 pins are required
+  to interface. The device's I2C address is either 0x76 or 0x77.
+
+  Adafruit invests time and resources providing this open source code,
+  please support Adafruit andopen-source hardware by purchasing products
+  from Adafruit!
+
+  Written by Limor Fried & Kevin Townsend for Adafruit Industries.
+  BSD license, all text above must be included in any redistribution
+  See the LICENSE file for details.
+ ***************************************************************************/
+
+#include <SmingCore.h>
+#include <Adafruit_BME280.h>
+#include <Wire.h>
+
+#define SEALEVELPRESSURE_HPA (1013.25)
+
+Adafruit_BME280 bme;
+Timer procTimer;
+
+// Will use default pins for selected architecture. You can override values here
+// #define SDA 4
+// #define SCL 5
+
+void printValues()
+{
+	Serial.print(F("Temperature = "));
+	Serial.print(bme.readTemperature());
+	Serial.println(" °C");
+
+	Serial.print(F("Pressure = "));
+
+	Serial.print(bme.readPressure() / 100.0F);
+	Serial.println(" hPa");
+
+	Serial.print(F("Approx. Altitude = "));
+	Serial.print(bme.readAltitude(SEALEVELPRESSURE_HPA));
+	Serial.println(" m");
+
+	Serial.print(F("Humidity = "));
+	Serial.print(bme.readHumidity());
+	Serial.println(" %");
+
+	Serial.println();
+}
+
+void init()
+{
+	Serial.begin(SERIAL_BAUD_RATE); // 115200 by default
+	Serial.systemDebugOutput(true); // Enable/disable debug output
+
+	Serial.println(F("BME280 test"));
+
+#ifdef SDA
+	Wire.pins(SDA, SCL);
+#endif
+
+	if(!bme.begin()) {
```

Comment:
suggestion add this comment to the line, it can help beginners to discover this possibility
`if(!bme.begin() {  // if(!bme.begin(0x76, &Wire)) if you need a specific address`